.github/workflows/build-and-test.yml

```diff
@@ -103,6 +129,7 @@ jobs:
   frontend-tests:
     name: Run Frontend Tests
     runs-on: ubuntu-latest
+    if: needs.check-diff.outputs.frontend == 'true'
```

Comment:
I think this block needs a `needs: check-diff`? 

Alternately, we could make this block have a `needs: build-frontend`, since there's not much point in running the frontend tests if the app doesn't even build. And then in that case, the `needs: check-diff` in `build-frontend` will halt this one's execution as well.

Reply:
I like that idea! Will update with `needs: build-frontend`